catalog/core/catalog-core-api/src/main/java/ddf/catalog/service/TransformerService.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.service;
+
+import ddf.catalog.transform.MultiInputTransformer;
+import ddf.catalog.transform.MultiMetacardTransformer;
+import ddf.catalog.transform.QueryResponseTransformer;
+import java.util.List;
+import javax.activation.MimeType;
+
+public interface TransformerService {
```

Comment:
I'm hesitant to pull more logic into TransformerService. I would rather see us create another class that replaces all of the places that perform that uses the transformers. I'm also concerned about adding more development time to this ticket.

Reply:
I think you are already touching most of the classes that brute force the transform.  If you look near the locators you can see the duplication.  I still think it would be worth it to see how much effort it would take to pull that duplicated code in so it is not the responsibility of everyone else to find the right transformer to use. 

Another issue is that it is easy to compromise the security of the system depending on how streams and files are used during transform.  Eventually we will need a centralized place to do transforms that focuses on security and improves maintainability.